### ISSUE TITLE
Suggestion:directly declared enter class and leave class in transition definition

### ISSUE THREAD

**Issue by bluedarker** [vuejs/vue#1956]
##### when using other lib like animate.css, can't integration easy with Vue.

> expect this 

```
<div v-show="showFlag" transition="transitionId:transitionEnter,transitionLeave">
```

> for integration with animate.css

```
<div v-show="showFlag" transition="transitionId:bounceInRight,bounceOutRight">
```

**Comment by yyx990803** [comment]
Sounds like a useful option...

How about we define it in a transition object:

``` js
Vue.transition('bounce', {
  enterClass: 'bounceInRight',
  leaveClass: 'bounceOutRight'
})
```

``` html
<div v-show="showFlag" transition="bounce">
```

**Comment by bluedarker** [comment]
Of course, but use a little complicated.

**Comment by yyx990803** [comment]
Alternatively:

``` html
<div v-show="showFlag" :transition="{ enterClass: 'bounceInRight', leaveClass: 'bounceOutRight' }">
```

**Comment by bluedarker** [comment]
Great, enough for me. but how provide js hook?

**Comment by yyx990803** [comment]
``` html
<div v-show="showFlag" :transition="{
  name: 'bounce',
  enterClass: 'bounceInRight',
  leaveClass: 'bounceOutRight'
}">
```

You can define `enterClass` and `leaveClass` either inline, or in the JavaScript definition.

**Comment by dsonet** [comment]
How about enterClass and leaveClass can also pass as function, since some times need random transitions or any other needs custom logic situation.

**Comment by OEvgeny** [comment]
@dsonet computed variables probably should work.

**Comment by dsonet** [comment]
@OEvgeny Thanks for your reply, I didn't try the way you mentioned yet, but it seems has to be inline. What I want is also can do that with Vue.transition to make it reusable.

**Comment by ediamin** [comment]
@yyx990803 Do you have any plan to make enterClass and leaveClass dynamic? For example, in a carousel, a slide may slide from both left and right. If I assign a vue transition to a slide, its enter and leave classes will be unchangeable. It will slide from either left OR right. But in this case, we need to assign dynamic classes.

By the way, thanks for the great work!

**Comment by Skhmt** [comment]
Will this be implemented in 2.0.0? I see it works in 1.0.14 and up, but not in 2.0.0 alpha.

**Comment by yyx990803** [comment]
This is supported in 1.0.14+ and 2.0 (as props to `<transition>`)
